test/core/utils/contains.js

```diff
@@ -40,95 +71,112 @@ describe('axe.utils.contains', function () {
+        `<section> <img> </section>`
+      );
+      const tree = axe.setup(fixture);
+      const node1 = axe.utils.querySelectorAll(tree, 'section')[0];
+      const node2 = axe.utils.querySelectorAll(tree, 'img')[0];
       assert.isTrue(axe.utils.contains(node1, node2));
-    }
-  );
-
-  (shadowSupported ? it : xit)(
-    'should work with slotted elements inside shadow DOM',
-    function () {
-      var tree, node1, node2;
-
-      function createContentSlotted() {
-        var group = document.createElement('div');
-        group.innerHTML = '<div id="target">Stuff<slot></slot></div>';
-        return group;
-      }
-      function makeShadowTree(node) {
-        var root = node.attachShadow({ mode: 'open' });
-        var div = document.createElement('div');
-        var a = document.createElement('a');
-        div.appendChild(a);
-        root.appendChild(div);
-        div.appendChild(createContentSlotted());
-      }
-      fixture.innerHTML = '<div></div>';
-      makeShadowTree(fixture.firstChild);
-      tree = axe.utils.getFlattenedTree(fixture.firstChild)[0].children;
-      node1 = axe.utils.querySelectorAll(tree, '#target')[0];
-      node2 = axe.utils.querySelectorAll(tree, 'a')[0];
+      assert.isFalse(axe.utils.contains(node2, node1));
+    });
+
+    it('works when the nodes are in nested trees', () => {
+      createNestedShadowDom(
+        fixture,
+        `<section id="shadowHost"></section>`,
+        `<div> <img> </div>`
+      );
+      const tree = axe.setup(fixture);
+      const node1 = axe.utils.querySelectorAll(tree, 'section')[0];
+      const node2 = axe.utils.querySelectorAll(tree, 'img')[0];
       assert.isTrue(axe.utils.contains(node1, node2));
-    }
-  );
-
-  it('should work', function () {
-    fixture.innerHTML = '<div id="outer"><div id="inner"></div></div>';
-    var inner = axe.utils.getFlattenedTree(document.getElementById('inner'))[0];
-    var outer = axe.utils.getFlattenedTree(document.getElementById('outer'))[0];
+      assert.isFalse(axe.utils.contains(node2, node1));
+    });
+
+    it('works when the nodes are in nested multiple layers deep', () => {
+      createNestedShadowDom(
+        fixture,
+        `<main id="shadowHost"></main>`,
+        '<article> <div id="shadowHost"></div> </article>',
+        '<section> <div id="shadowHost"></div> </section>',
+        `<div> <img> </div>`
+      );
+      const tree = axe.setup(fixture);
+      const node1 = axe.utils.querySelectorAll(tree, 'main')[0];
+      const node2 = axe.utils.querySelectorAll(tree, 'img')[0];
+      assert.isTrue(axe.utils.contains(node1, node2));
+      assert.isFalse(axe.utils.contains(node2, node1));
+    });
+
+    it('is false when the nodes are in adjacent trees', () => {
+      const section = createNestedShadowDom(
+        fixture,
+        `<section id="target"></section>`
+      );
+      createNestedShadowDom(section, `<img>`);
+      createNestedShadowDom(section, `<input>`);
```

Comment:
Did you mean for this to not have any shadow trees? Passing a single html code to `createNestedShadowDom` just appends the code to `innerHTML` and never creates a shadow tree, so the end result is just flat nodes:

```html
<div id="fixture">
  <section id="target">
    <img>
    <input>
  </section>
</div>
```

Reply:
Ooff! Good catch. This test is completely broken. Will fix.